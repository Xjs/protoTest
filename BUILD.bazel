load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")

proto_library(
    name = "a_proto",
    srcs = ["foo/x.proto"],
    visibility = ["//visibility:public"],
)

go_proto_library(
    name = "a_go_proto",
    importpath = "github.com/Xjs/protoTest/a",
    proto = ":a_proto",
    visibility = ["//visibility:public"],
)

proto_library(
    name = "b_proto",
    srcs = ["foo/b.proto"],
    visibility = ["//visibility:public"],
)

go_proto_library(
    name = "b_go_proto",
    importpath = "github.com/Xjs/protoTest/b",
    proto = ":b_proto",
    visibility = ["//visibility:public"],
)

go_library(
    name = "go_default_library",
    srcs = ["foo.go"],
    importpath = "main",
    visibility = ["//visibility:private"],
    deps = [
        ":a_go_proto",
    ],
)

go_binary(
    name = "main",
    embed = [":go_default_library"],
    visibility = ["//visibility:public"],
)
