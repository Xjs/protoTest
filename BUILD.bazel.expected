load("@io_bazel_rules_go//go:def.bzl", "go_library")
load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")

proto_library(
    name = "a_proto",
    srcs = ["a.proto"],
    visibility = ["//visibility:public"],
)

go_proto_library(
    name = "a_go_proto",
    importpath = "github.com/Xjs/protoTest",
    proto = ":a_proto",
    visibility = ["//visibility:public"],
)

proto_library(
    name = "b_proto",
    srcs = ["b.proto"],
    visibility = ["//visibility:public"],
)

go_proto_library(
    name = "b_go_proto",
    importpath = "github.com/Xjs/protoTest",
    proto = ":b_proto",
    visibility = ["//visibility:public"],
)

go_library(
    name = "a",
    importpath = "github.com/Xjs/protoTest/a",
    library = ":a_go_proto",
    visibility = ["//visibility:public"],
)

go_library(
    name = "b",
    importpath = "github.com/Xjs/protoTest/b",
    library = ":b_go_proto",
    visibility = ["//visibility:public"],
)


